File: DeepSpaceJava/src/WeaponType.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 *
 * @author Samuel Medina
 * @author Laura Sánchez 
 */

public enum WeaponType {
    /* Podría necesitar arreglos, a la espera del correo de prácticas*/
    
    LASER((float) 2.0), MISSILE((float) 3.0), PLASMA((float) 4.0);
    
    private float power;
    
    WeaponType(float p){
        this.power = p;
    }
    
    float getPower(){
        return power;
    }
    
}

File: DeepSpaceRuby/lib/WeaponType.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
module Deepspace
  module WeaponType
    class Type
      
      def initialize(p,n)
        @power=p
        @name=n
      end

      def power
        @power
      end
      
      def name 
        @name
      end
      
      def to_s
        @power.to_s + " " + @name.to_s
      end

    end
    
      LASER=Type.new(2.0, "LASER")
      MISSILE=Type.new(3.0, "MISSILE")
      PLASMA=Type.new(4.0, "PLASMA")
  end
end